extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -146,4 +152,41 @@ declare module 'dataworkspace' {
 	export interface IProjectActionGroup {
 		actions: IProjectAction[];
 	}
+
+	/**
+	 * Defines table to be presented in the dashboard container
+	 */
+	export interface IDashboardTable {
+		/**
+		 * name of the table
+		 */
+		name: string;
+
+		/**
+		 * column header information
+		 */
+		columns: IDashboardColumnInfo[];
+
+		/**
+		 * data that goes in the table
+		 */
+		data: (string | DashboardTableData)[][];
+	}
+
+	/**
+	 * Header information of a column in the table
+	 */
+	export interface IDashboardColumnInfo {
+		displayName: string;
+		width: number;
+		valueType: azdata.DeclarativeDataType;
```

Comment:
here is the definition of the DeclarativeDataType, the table in the dashboard will be read-only, I think we don't really need to expose the valueType here. you can auto determine the type, if icon then component, otherwise string.
```
export enum DeclarativeDataType {
		string = 'string',
		category = 'category',
		boolean = 'boolean',
		editableCategory = 'editableCategory',
                component = 'component'
	}
```

Reply:
Changed to an enum instead of DeclarativeDataType.